packages/popups/src/UIPushNotice/UIPushNoticeContent.tsx

```diff
@@ -0,0 +1,127 @@
+import * as React from 'react';
+import { useWindowDimensions, StyleSheet } from 'react-native';
+import Animated, { useAnimatedStyle } from 'react-native-reanimated';
+
+import type { UIImageProps } from '@tonlabs/uikit.hydrogen';
+import {
+    useShadow,
+    useTheme,
+    UIBackgroundView,
+    UIBackgroundViewColors,
+    UILabel,
+    UILabelRoles,
+    UIImage,
+    makeStyles,
+} from '@tonlabs/uikit.hydrogen';
+import { UIConstant } from '@tonlabs/uikit.navigation';
+
+import type { useNoticeVisibility } from '../Notice/hooks/useNoticeVisibility';
+
+export type UIPushNoticeContentPublicProps = {
+    title: string;
+    message: string;
+    icon?: UIImageProps['source'];
+    hasCountdown?: boolean;
+};
+
+type ContentPrivateProps = {
+    onPress: () => void;
+    onLongPress: () => void;
+    onPressOut: () => void;
+    countdownProgress: ReturnType<typeof useNoticeVisibility>['countdownProgress'];
+};
+
+function LineCountdown({
+    countdownProgress,
+}: {
+    countdownProgress: ContentPrivateProps['countdownProgress'];
+}) {
+    const { width } = useWindowDimensions();
+
+    const noticeWidth = React.useMemo(
+        () => width - UIConstant.notice.toastIndentFromScreenEdges * 2,
+        [width],
+    );
+
+    const countdownStyle = useAnimatedStyle(() => {
+        return {
+            transform: [
+                {
+                    translateX: countdownProgress.value * noticeWidth * -1,
+                },
+            ],
+        };
+    }, [noticeWidth]);
+
+    const theme = useTheme();
+    const dynamicStyles = useStyles(theme);
+    return (
+        <UIBackgroundView style={styles.countdownContainer}>
+            <Animated.View style={[dynamicStyles.countdownLine, countdownStyle]}></Animated.View>
+        </UIBackgroundView>
+    );
+}
+
+const useStyles = makeStyles((theme: ReturnType<typeof useTheme>) => ({
```

Comment:
`ReturnType<typeof useTheme>` - there is a ready-made type `Theme` from '@tonlabs/uikit.hydrogen'